### PR TITLE
feat: PR check and initial test

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Initial PR check with linting, formatting, and tests.

May need a couple attempts, but I think this has all the basics.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Part of #36.

### How to test this PR?

Hopefully, it's the first PR that tests itself. ;-)

### CODE REVIEW

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+jobs:
+  lint-format-unit:
+    name: linter, formatter, and tests / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-2022, ubuntu-22.04, macos-11]
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:30]:
```suggestion
        os: [windows-2022, ubuntu-22.04, macos-12]
```

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        with:
+          node-version: 18
+
+      - name: Get yarn cache directory path (Windows)
+        if: ${{ matrix.os=='windows-2022' }}
+        id: yarn-cache-dir-path-windows
+        run: echo "dir=$(yarn cache dir)" >> ${env:GITHUB_OUTPUT}
+
+      - name: Get yarn cache directory path (mac/Linux)
+        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-11' }}
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:43]:
```suggestion
        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-12' }}
```

File: .github/workflows/pr-check.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if: ${{ matrix.os=='windows-2022' }}
+        id: yarn-cache-windows
+        with:
+          path: ${{ steps.yarn-cache-dir-path-windows.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - uses: actions/cache@v4
+        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-11' }}
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:57]:
```suggestion
        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-12' }}
```

File: .github/workflows/pr-check.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+      - name: Execute yarn
+        run: yarn --frozen-lockfile --network-timeout 180000
+
+      - name: Run linter
+        run: yarn lint:check
+
+        # skip formatter on windows
+      - name: Run formatter
+        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-11' }}
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:73]:
```suggestion
        if: ${{ matrix.os=='ubuntu-22.04' || matrix.os=='macos-12' }}
```

**Review by benoitf — APPROVED** [review]

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (deboer-tim) [reply .github/workflows/pr-check.yaml]: Updated in new commit.

File: .github/workflows/pr-check.yaml (hunk 4 above)
Comment (deboer-tim) [reply .github/workflows/pr-check.yaml]: Updated in new commit.

File: .github/workflows/pr-check.yaml (hunk 3 above)
Comment (deboer-tim) [reply .github/workflows/pr-check.yaml]: Updated in new commit.

File: .github/workflows/pr-check.yaml (hunk 2 above)
Comment (deboer-tim) [reply .github/workflows/pr-check.yaml]: Updated in new commit.

**Pull request merged**